File: Tree/100. Same Tree/100. Same Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

// 检验是否为相同的树的方式较为简单，一般使用递归方法：树root1和树root2相等，当且仅当根节点相等，左子树和右子树对应相等
// 附注：使用输出空结点的遍历也可以验证两个树相等，在这里略去

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(!p&&!q)
        return true;
        else if(p&&!q||q&&!p)
        return false;
        else
        {
            if(isSameTree(p->left,q->left)&&isSameTree(p->right,q->right)&&p->val==q->val)
            return true;
            else
            return false;
        }
    }
};

File: Tree/100. Same Tree/100. Same Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

# 本题的难度较小，可以直接使用递归进行求解，二叉树root1和root2相等，只需要根节点相等，左子树和右子树分别相等即可

class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """

        if not p and not q:
            return True
        elif not p and q:
            return False
        elif not q and p:
            return False
        else:
            if self.isSameTree(p.left,q.left) and self.isSameTree(p.right,q.right) and p.val==q.val:
                return True
            else:
                return False


        